crates/base-db/src/input.rs

```diff
@@ -304,6 +304,7 @@ pub struct CrateData {
     /// For purposes of analysis, crates are anonymous (only names in
     /// `Dependency` matters), this name should only be used for UI.
     pub display_name: Option<CrateDisplayName>,
+    pub crate_root_path: Option<AbsPathBuf>,
```

Comment:
We shouldn't store any paths like this in `CrateData`, instead we should resolve the `root_file_id` back to its path where we need it

Reply:
Ok, I was unsure that this was the right aproach, but could you elaborate a little more on why this is bad?

Reply:
We want as little code to know about paths as possible. Paths are tricky to handle correctly as different paths can describe the same file etc, FileId's are unambiguous. And in this case, we already store the file id, so storing the path as well is redundant.